src/server_manager/web_app/app.ts

```diff
@@ -289,319 +312,197 @@ export class App {
       this.appRoot.openGetConnectedDialog(this.getS3InviteUrl(event.detail.accessKey, true));
     });
 
-    appRoot.addEventListener('ShowServerRequested', (event: CustomEvent) => {
-      this.handleShowServerRequested(event.detail.displayServerId);
+    appRoot.addEventListener('ShowServerRequested', async (event: CustomEvent) => {
+      const server = this.getServerById(event.detail.displayServerId);
+      if (server) {
+        await this.showServer(server);
+      } else {
+        console.error(
```

Comment:
If we manage the state properly, I think this should never happen, but it might be good to be defensive. 

Should we display an error notification to the user (and clean up state, like removing it from the localServerIds cache and UI) when this happens?

Reply:
You are right, this should never happen. I added a comment clarifying.
I'm not sure about displaying an error because I don't really know in what situation this would happen.